docs/content/guides/developer/sui-101/shared-owned.mdx

```diff
@@ -1,20 +1,171 @@
+
+```rust
+public fun create<T: key + store>(
+    escrowed: T,
+    exchange_key: ID,
+    recipient: address,
+    ctx: &mut TxContext
+) {
+    let escrow = Escrow {
+        id: object::new(ctx),
+        sender: tx_context::sender(ctx),
+        recipient,
+        exchange_key,
+        escrowed: option::some(escrowed),
+    };
+
+    transfer::public_share_object(escrow);
+}
+```
+
+This time the swap request is a shared object, and accepts the object being escrowed directly (not locked).  The request remembers the address that sent it (who is allowed to reclaim the object if the swap hasn't already happened), and the intended recipient, who is then expected to continue the swap by providing the object they initially locked:
+
+```rust
+public fun swap<T: key + store, U: key + store>(
+    escrow: &mut Escrow<T>,
+    key: Key,
+    locked: Locked<U>,
+    ctx: &TxContext,
+): T {
+    assert!(option::is_some(&escrow.escrowed), EAlreadyExchangedOrReturned);
+    assert!(escrow.recipient == tx_context::sender(ctx), EMismatchedSenderRecipient);
+    assert!(escrow.exchange_key == object::id(&key), EMismatchedExchangeObject);
+
+    let escrowed1 = option::extract<T>(&mut escrow.escrowed);
+    let escrowed2 = lock::unlock(locked, key);
+
+    // Do the actual swap
+    transfer::public_transfer(escrowed2, escrow.sender);
+    escrowed1
+}
+```
+
+Even though the `Escrow` request is a shared object that is accessible by anyone, the Move interface ensures that only the original sender and the intended recipient can successfully interact with it. `swap` checks that locked object matches the object that was requested when the `Escrow` was created (again, by comparing key IDs) and assumes that the intended recipient wants the escrowed object (if they did not, they would not have called `swap`).
+
+Assuming all checks pass, the object held in `Escrow` is extracted and returned to the first party, leaving the shared request object empty (but still existing).  The locked object offered by the first party is also unlocked and sent to the second party, completing the swap.
+
+### Comparison
+
+We have seen two ways to implement a swap between two objects.  In both cases there is a point at which one party has made a request and the other has not responded.  At this point, both parties may want to access the `Escrow` object: One to cancel the swap, and the other to complete it.
+
+In one case, the protocol uses only owned objects but requires a custodian to act as an intermediary. This has the advantage of avoiding the costs and latencies of consensus altogether, but involves more steps and requires trusting a third party for liveness.
+
+In the other case, the object is custodied on chain in a shared object.  This requires consensus, and leaves a shared object on-chain at the end of the swap (consuming gas in the form of irretrievable storage rebate), but involves fewer steps, and no third party.
```

Comment:
> leaves a shared object on-chain at the end of the swap.

Hopefully not for long!  cc @tzakian to update this example and the docs once shared object deletion hits the production networks.